tests/integration/test_kube_ovn.py

```diff
@@ -51,3 +60,180 @@ async def test_kubectl_ko_plugin(ops_test: OpsTest):
         assert (
             rc == 0
         ), f"Failed to execute kubectl-ko on machine:{m} {(stderr or stdout).strip()}"
+
+
+async def test_pod_network_limits(ops_test, kubeconfig, client, iperf3_yaml_path):
+    with open(iperf3_yaml_path) as f:
+        for obj in codecs.load_all_yaml(f):
+            client.create(obj)
+
+    wait_daemonset(client, "ls1", "perf", 3)
+
+    pods = list(client.list(Pod, namespace="ls1"))
+
+    for pod in pods:
+        client.wait(
+            Pod,
+            pod.metadata.name,
+            for_conditions=["Ready"],
+            namespace=pod.metadata.namespace,
+        )
+
+    server, test_pod, _ = pods
+    namespace = server.metadata.namespace
+
+    log.info("Annotate test pod with rate limit values...")
+    rate_values = (
+        'ovn.kubernetes.io/ingress_rate="10" ovn.kubernetes.io/egress_rate="5"'
+    )
+    cmd = (
+        f"kubectl --kubeconfig {kubeconfig} "
+        "annotate --overwrite "
+        f"pod {test_pod.metadata.name} "
+        f"-n {namespace} {rate_values}"
+    )
+    rc, stdout, stderr = await ops_test.run(*shlex.split(cmd))
```

Comment:
you can annotate with lightkube library.  

it seems annotations are a part of the metadata object of the pod.  But i get this is just so much faster. :)